hoth_api.yaml

```diff
@@ -319,5 +319,351 @@ components:
+        usage_description:
+          type: string
+          maxLength: 1000
+        need_for_entire_TO_duration:
+          type: boolean
+        applicable_periods:
+          type: array
+          items:
+            $ref: "#/components/schemas/Period"
+        service_offerings: # represents the "Select Service Offerings" field
+          type: array
+          items:
+            $ref: "#/components/schemas/ServiceOffering"
+        other_service_offering:
+          type: string
+          maxLength: 100
+    Period:
+      description: "Part of DOW, represents Period Table fields"
+      properties:
+        period_type:
+          type: string
+          enum:
+            - Base
+            - Option
+        period_unit_count:
+          type: integer
+        period_unit:
+          type: string
+          enum:
+            - Day(s)
+            - Week(s)
+            - Month(s)
+            - Year(s)
+        option_order:
+          type: integer
+    ServiceOffering:
+      description: "Part of DOW, represents a single Service Offering (Service Offering table)"
+      properties:
+        service_offering_group:
+          type: string
+          enum:
+            - Advanced Technology and Algorithmic Techniques (Machine Learning)
+            - Advisory and Assistance
+            - Applications
+            - Compute
+            - Database with Storage
+            - Developer Tools and Services
+            - Edge Computing and Tactical Edge (TE)
+            - General IaaS, PaaS, and SaaS
+            - Internet of Things (IoT)
+            - Networking
+            - Security
+            - Training
+        name:
+          type: string
+          maxLength: 100
+    PeriodOfPerformance:
+      description: "Part of DOW, represents a single Period of Performance"
+      properties:
+        base_period:
+          $ref: "#/components/schemas/Period"
+        option_periods:
+          type: array
+          items:
+            $ref: "#/components/schemas/Period"
+          maxItems: 4000
+        pop_start_request:
+          type: string
+          enum:
+            - Unselected
+            - Yes
+            - No
+        requested_pop_start_date:
+          type: string
+          format: date
+          example: "2021-07-01"
+        time_frame:
+          type: string
+          enum:
+            - No later than
+            - No sooner than
+        recurring_requirement:
+          type: string
+          enum:
+            - Unselected
+            - Yes
+            - No
+    GFEOverview:
+      description: "Part of DOW, represents GFE Overview fields"
+      properties:
+        dpas_unit_id:
+          type: string
+          maxLength: 6
+        property_custodian_name:
+          type: string
+          maxLength: 40 # This is another instance of maxLength in SNOW being weird, the person might have a name longer than 40 characters
```

Comment:
So for _all_ of these `maxLength` things, does the Document Generation API _care_ at all? I mean, let's not use this to encode limitations of downstream systems. If we can support a system with 41 characters or 1000 characters then let's stop putting `maxLength: N` just because "that's what SNOW does here". Every `maxLength` should encode _validation_ data or a true limitation of our API. So for things like ID Numbers etc that we know have a specific length, then `maxLength` (and potentially a `minLength`) are very valuable. If this is a person's name, we absolutely must not limit it unless for some reason we'd fail to generate a document (or if we'd generate something invalid) with more characters or if >40char is an indication of an error.